File: Calificaciones con FOR VB/Calificaciones con FOR VB/Form1.vb
Public Class frmCalificacionesFORVB

    Dim numCantidad As Integer
    Dim numInput As Double
    Dim numProm As Double
    Dim numEntrada As String


    Private Sub txtCantidad_KeyPress(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles txtCantidad.KeyPress

        If txtCantidad.Text = String.Empty Then

            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "." And Not e.KeyChar = "-"

        ElseIf Not txtCantidad.Text.Contains(".") Then

            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "."

        ElseIf Not txtCantidad.Text.Contains("-") Then

            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "-"

        ElseIf txtCantidad.Text.Contains(".") And txtCantidad.Text.Contains("-") Then

            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar)

        End If

    End Sub

    Private Sub btnOK_Click(sender As Object, e As EventArgs) Handles btnOK.Click
        If ((txtCantidad.Text IsNot String.Empty) And (txtCantidad.Text IsNot String.Empty)) Then

            If txtCantidad.Text Like "*.*-*" Or txtCantidad.Text Like "0" Or txtCantidad.Text Like "*-*" Or txtCantidad.Text Like "*.*" Or txtCantidad.Text Like "-" Or txtCantidad.Text Like "-." Then

                MsgBox("Valor invalido")
                Exit Sub

            End If

            numInput = 0
            Dim notas(txtCantidad.Text - 1) As String
            numCantidad = Convert.ToInt32(txtCantidad.Text)





            For Each nota As Integer In notas



                numEntrada = InputBox("Calificacion (base 10)" & ": ")
                numInput = Val(numEntrada)



                If IsNumeric(numEntrada) Then

                    If numEntrada > 10 Then

                        MsgBox("Valores invalidos")
                        Exit Sub
                    End If

                    notas(nota) = numInput



                Else

                    MsgBox("Valores invalidos")
                    Exit Sub

                End If

                numProm += notas(nota)

            Next

            numProm = (numProm / numCantidad)
            MsgBox(numProm)

            Select Case numProm
                Case 10
                    MsgBox("AU")
                Case 9 To 10
                    MsgBox("DE")
                Case 8 To 9
                    MsgBox("SA")
                Case Else
                    MsgBox("NA")
            End Select

        End If

        numProm = 0

    End Sub
End Class

File: Calificaciones con FOR CS/CalificacionesForCS/CalificacionesForCS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace CalificacionesForCS
{
	public partial class Form1 : Form
	{
		public Form1()
		{
			InitializeComponent();
		}



		private int numCantidad;
		private double numInput;
		private double numProm;
		private string numEntrada;


		private void txtCantidad_KeyPress(object sender, KeyPressEventArgs e)
		{
			if ((txtCantidad.Text ?? "") == (string.Empty ?? ""))
			{
				e.Handled = !char.IsDigit(e.KeyChar) & !char.IsControl(e.KeyChar) & !(Conversions.ToString(e.KeyChar) == ".") & !(Conversions.ToString(e.KeyChar) == "-");
			}
			else if (!txtCantidad.Text.Contains("."))
			{
				e.Handled = !char.IsDigit(e.KeyChar) & !char.IsControl(e.KeyChar) & !(Conversions.ToString(e.KeyChar) == ".");
			}
			else if (!txtCantidad.Text.Contains("-"))
			{
				e.Handled = !char.IsDigit(e.KeyChar) & !char.IsControl(e.KeyChar) & !(Conversions.ToString(e.KeyChar) == "-");
			}
			else if (txtCantidad.Text.Contains(".") & txtCantidad.Text.Contains("-"))
			{
				e.Handled = !char.IsDigit(e.KeyChar) & !char.IsControl(e.KeyChar);
			}
		}

		private void btnOK_Click(object sender, EventArgs e)
		{

			if ((!string.IsNullOrEmpty(txtCantidad.Text)) && (!string.IsNullOrEmpty(txtCantidad.Text)))
			{

				if (LikeOperator.LikeString(txtCantidad.Text, "*.*-*", CompareMethod.Binary) | LikeOperator.LikeString(txtCantidad.Text, "0", CompareMethod.Binary) | LikeOperator.LikeString(txtCantidad.Text, "*-*", CompareMethod.Binary) | LikeOperator.LikeString(txtCantidad.Text, "*.*", CompareMethod.Binary) | LikeOperator.LikeString(txtCantidad.Text, "-", CompareMethod.Binary) | LikeOperator.LikeString(txtCantidad.Text, "-.", CompareMethod.Binary))
				{
					MessageBox.Show("Valor invalido");
					return;
				}


				numInput = 0;
				var notas = new string[(int)(Conversion.Val(txtCantidad.Text) - 1) + 1];
				numCantidad = Convert.ToInt32(txtCantidad.Text);


				foreach (var nota in notas)
				{
					numEntrada = Interaction.InputBox("Calificacion (base 10)" + ": ");
					numInput = Conversion.Val(numEntrada);
					if (Information.IsNumeric(numEntrada))
					{
						if (Conversions.ToDouble(numEntrada) > 10)
						{
							MessageBox.Show("Valores invalidos");
							return;
						}

						notas[Convert.ToInt32(nota)] = Convert.ToString(numInput);
					}
					else
					{
						MessageBox.Show("Valores invalidos");
						return;
					}

					numProm += Conversions.ToDouble(notas[Convert.ToInt32(nota)]);



					
				}

				numProm = numProm / numCantidad;
				MessageBox.Show(Convert.ToString(numProm));
				var switchExpr = numProm;
				switch (switchExpr)
				{
					case 10:
						{
							MessageBox.Show("AU");
							break;
						}

					case object _ when 9 <= switchExpr && switchExpr <= 10:
						{
							MessageBox.Show("DE");
							break;
						}

					case object _ when 8 <= switchExpr && switchExpr <= 9:
						{
							MessageBox.Show("SA");
							break;
						}

					default:
						{
							MessageBox.Show("NA");
							break;
						}
				}

				
			}
			numProm = 0;
		}
	}
}
